### PR TITLE
dsp.c: Fix and improve potentially inaccurate log message.

### PR DESCRIPTION
**Pull request by InterLinked1**
If ast_dsp_process is called with a codec besides slin, ulaw, or alaw, a warning is logged that in-band DTMF is not supported, but this message is not always appropriate or correct, because ast_dsp_process is much more generic than just DTMF detection.

This logs a more generic message in those cases, and also improves codec-mismatch logging throughout dsp.c by ensuring incompatible codecs are printed out.

Resolves: #595

### CODE REVIEW

**Comment by InterLinked1** [comment]
cherry-pick-to: 18
cherry-pick-to: 20
cherry-pick-to: 21

File: main/dsp.c [hunk 1]
```diff
@@ -429,7 +429,7 @@ struct ast_dsp {
 	int faxmode;
 	int freqmode;
 	int dtmf_began;
-	int display_inband_dtmf_warning;
+	unsigned int display_inband_dtmf_warning:1;
```
Comment (seanbright) [inline main/dsp.c:432]: Unrelated change that snuck in? If not can you explain the rationale for this change?

File: main/dsp.c [hunk 2]
```diff
@@ -1223,7 +1223,7 @@ int ast_dsp_call_progress(struct ast_dsp *dsp, struct ast_frame *inf)
 		return 0;
 	}
 	if (!ast_format_cache_is_slinear(inf->subclass.format)) {
-		ast_log(LOG_WARNING, "Can only check call progress in signed-linear frames\n");
+		ast_log(LOG_WARNING, "Can only check call progress in signed-linear frames, %s not supported\n", ast_format_get_name(inf->subclass.format));
```
Comment (seanbright) [inline main/dsp.c:1226]: These lines are pretty long now. I might look nicer to add a line break after the format string. This applies to all of the logging statement changes you made in this file.

**Review by seanbright — CHANGES_REQUESTED** [review]

File: main/dsp.c (hunk 1 above)
Comment (InterLinked1) [reply main/dsp.c:432]: It is related; I reviewed all the code where this is used and set, and it only needs to be a bitfield, not an int.
I considered renaming the variable too for clarity but guess I didn't do that.

Comment (seanbright) [reply main/dsp.c:432]: But you aren't changing any code that reads or sets this structure member, so it's not clear to me what the benefit is. Does it change the generated code? Due to alignment concerns I don't think it would change the size of the struct either?

Comment (InterLinked1) [reply main/dsp.c:432]: It's more semantic. You can look at the struct and know that it's a boolean variable, which was not conveyed before. I think it helps with code clarity to do this, regardless of space savings, since it serves as documentation.

If more bitfields are added in the future, then it would actually reduce memory usage.

Comment (seanbright) [reply main/dsp.c:432]: > If more bitfields are added in the future, then it would actually reduce memory usage.

I don't think that is correct, but I will defer to your expertise on this.

Comment (jcolp) [reply main/dsp.c:432]: I don't like change for the sake of change. If that time ever comes, then it can be changed at that point. Right now it is premature to touch it or care about it. Just changing the log message is fine.

**Review by jcolp — CHANGES_REQUESTED** [review]

File: main/dsp.c (hunk 1 above)
Comment (InterLinked1) [reply main/dsp.c]: It wasn't really for a future hypothetical performance benefit, it was more for making the code more self-documenting, since that variable is already a misnomer. I've changed it back but added a documentation note about this.

**Review by jcolp — APPROVED** [review]

File: main/dsp.c (hunk 1 above)
Comment (seanbright) [reply main/dsp.c]: Looks like `busymaybe` could be a bitfield too. I would add the same comment there so people aren't confused.

**Review by jcolp — APPROVED** [review]
